### PR TITLE
Stricter/smaller types for some internal BetterReflection symbols

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: src/Reflection/ReflectionClass.php [hunk 1]
```diff
@@ -68,7 +67,7 @@ class ReflectionClass implements Reflection
 
     private LocatedSource $locatedSource;
 
-    private ClassLikeNode $node;
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php:71]: Really like how we get rid of `ClassLikeNode` :+1:

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Comment by Ocramius** [comment]
Thanks @kukulich! Also, mutation score went up a bit :D

**Comment by kukulich** [comment]
> Also, mutation score went up a bit

Yes, that was my intention :)

**Pull request merged**